lib/src/sql/number.rs

```diff
@@ -198,48 +195,48 @@ impl Number {
 	pub fn is_integer(&self) -> bool {
 		match self {
 			Number::Int(_) => true,
-			Number::Float(_) => false,
-			Number::Decimal(v) => v.is_integer(),
+			Number::Float(v) => v.fract() == 0.0,
+			Number::Decimal(v) => v.scale() == 0,
 		}
 	}
```

Comment:
Note: I made it so floats return `true` here if they represent an exact integer, just like decimals do.